collector/perf_linux.go

```diff
@@ -244,6 +244,21 @@ func newPerfTracepointCollector(
 	return c, nil
 }
 
+// checkMaxOpenFileLimit returns an error if the getrlimit for RLIMIT_NOFILE is
+// likely too low.
+func checkMaxOpenFileLimit(cpus int, profilers int) error {
+	maxOpenFiles, err := perf.MaxOpenFiles()
```

Comment:
This function is just a wrapper around [`unix.Getrlimit`](https://pkg.go.dev/golang.org/x/sys/unix#Getrlimit).